resourcemanager/commonids/tenant.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package commonids
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+)
+
+var _ resourceids.ResourceId = &TenantId{}
+
+// TenantId is a struct representing the Resource ID for a Tenant
+type TenantId struct {
+	TenantId string
+}
+
+// DefaultTenantId returns a default TenantId
+var DefaultTenantId = NewTenantID("/")
+
+// NewTenantID returns a new TenantId struct
+func NewTenantID(tenantId string) TenantId {
+	return TenantId{
+		TenantId: tenantId,
+	}
+}
```

Comment:
Thinking about the default Tenant ID perhaps we want to handle this differently? If we make the `TenantId` field a pointer, we could support parsing both the Default Tenant ID and another Tenant ID? It'd be different to the other Resource ID parsers but this is a unique case where the Default is handled differently, which I don't believe we've got anywhere else - so perhaps it's worth handling that explicitly here?

```suggestion
// TenantId is a struct representing the Resource ID for a Tenant
type TenantId struct {
	TenantId *string
}

// NewDefaultTenantId returns a TenantId for the default Tenant
func NewDefaultTenantID() TenantId {
	return TenantId{}
}

// NewTenantID returns a new TenantId struct
func NewTenantID(tenantId string) TenantId {
	return TenantId{
		TenantId: &tenantId,
	}
}
```

WDYT @catriona-m @manicminer?